schema/schema_validator.py

```diff
@@ -36,74 +48,160 @@ def __init__(self):
+                file_path_pair = self.get_schema_data_info(icu_version, test_type)
+                if file_path_pair:
+                    schema_test_info.append(file_path_pair)
+
+        results = self.parallel_check_test_data_schema(schema_test_info)
+
+        for result_data in results:
+            logging.debug('test result data = %s', result_data)
+            msg = result_data['err_info']
+            if not result_data['data_file_name']:
+                # This is not an error but simple a test that wasn't run.
+                continue
+            if not result_data['result']:
+                logging.warning('FAIL: Test data %s, %s. MSG=%s',
+                                test_type, icu_version, result_data['err_info'])
+            else:
+                logging.info('Test data validated: %s %s', result_data['test_type'], result_data['icu_version'])
+            all_results.append(result_data)
         return all_results
 
+    def parallel_check_test_data_schema(self, schema_test_data):
+        num_processors = mp.cpu_count()
+        logging.info('Schema validation: %s processors for %s schema/test data pairs',
+                     num_processors,
+                     len(schema_test_data))
+
+        # Returns all the results
+        processor_pool = mp.Pool(num_processors)
+        with processor_pool as p:
+            result = p.map(self.check_test_data_against_schema, schema_test_data)
+        return result
+
+    def get_schema_data_info(self, icu_version, test_type):
+        # Gets pairs of schema and file names for test_type
+        schema_verify_file = os.path.join( self.schema_base, test_type, 'test_schema.json')
+        filename_map = SCHEMA_FILE_MAP[test_type]
+        result_file_name = SCHEMA_FILE_MAP[test_type]['test_data']['prod_file']
+        test_file_name = os.path.join(self.test_data_base, icu_version, result_file_name)
+        if os.path.exists(test_file_name):
+            return {
+                'test_type': test_type,
+                'icu_version': icu_version,
+                'schema_verify_file': schema_verify_file,
+                'test_result_file': test_file_name
+            }
+        else:
+            return None
+
+    def check_test_data_against_schema(self, schema_info):
+        icu_version = schema_info['icu_version']
+        test_type = schema_info['test_type']
+        schema_verify_file = schema_info['schema_verify_file'],
+        test_file_name = schema_info['test_result_file']
+        results = {
+            'test_type': test_type,
+            'icu_version': icu_version,
+            'result': None,
+            'err_info': None,
+            'test_schema': schema_verify_file,
+            'data_file_name': test_file_name
+        }
+        result = self.validate_json_file(schema_info)
+
+        if isinstance(result, list):
+            results['error_message'] = result[0]
+        else:
+            results['error_info'] = result['error']
```

Comment:
Observation: sometimes you use `error_info` and other times you use `err_info`.

Reply:
I'll examine this.